src/mechanics/mesh/MeshGmsh.h

```diff
@@ -0,0 +1,93 @@
+#pragma once
+
+#include "MeshFem.h"
+#include "base/Group.h"
+#include <string>
+#include <vector>
+#include <map>
+
+class GmshFileContent;
+
+namespace NuTo
+{
+
+//! @brief Reads a gmsh msh-file and converts it to a MeshFEM. Additionally physical groups are created and can be
+//! accessed via ID or name
+class MeshGmsh
+{
+public:
+    //! @brief ctor
+    //! @remark deleted: The class is always constructed with its associated files name
+    MeshGmsh() = delete;
```

Comment:
I would remove those two
~~~cpp
MeshGmsh() = delete;
~MeshGmsh() = delete;
~~~
because I find them pointless. But this may be personal preference.

Reply:
I would advice against deleting default dtors :stuck_out_tongue: - change it if you like, but I prefer to explicitly write down the big 5 + dtor so I don't have to guess what the compiler is doing. But as you said preferences.